lib/rules/no-unused-prop-types.js

```diff
@@ -1058,7 +989,7 @@ module.exports = {
       ObjectPattern: function(node) {
         // If the object pattern is a destructured props object in a lifecycle
         // method -- mark it for used props.
-        if (isNodeALifeCycleMethod(node.parent.parent)) {
+        if (astUtil.inLifeCycleMethod(context)) {
```

Comment:
is this necessarily equivalent? `context.getScope()` won't be the same as `node.parent.parent`, necessarily.